### ISSUE TITLE
Can't use scoped formio keys with variables

### ISSUE THREAD

**Issue by sergei-maertens** [open-formulieren/open-forms#1677]
I'm trying to set a `foo.bar` key for a component in Formio to test nesting for another issue, but this is blocked by the variable creation validation.

This is valid formio configuration, so `.` should be allowed in the key of a variable. (which probably complicates things).

Additionally, the backend endpoint errored on that nested/list validation error, so that'll need some attention too.

Validation error from console:

```
rest_framework.exceptions.ValidationError: [{}, {}, {}, {}, {}, {}, {}, {}, {'name': [ErrorDetail(string='Dit veld mag niet leeg zijn.', code='blank')], 'key': [ErrorDetail(string='Voer een geldige "slug" in, bestaande uit letters, cijfers, lage streepjes of streepjes.', code='invalid')]}, {'name': [ErrorDetail(string='Dit veld mag niet leeg zijn.', code='blank')], 'key': [ErrorDetail(string='Voer een geldige "slug" in, bestaande uit letters, cijfers, lage streepjes of streepjes.', code='invalid')]}]
```
